src/lib/apputils/net-server.c

```diff
@@ -1351,10 +1359,14 @@ process_tcp_connection_read(verto_ctx *ctx, verto_ev *ev)
         state->request.data = conn->buffer + 4;
 
         if (getsockname(verto_get_fd(ev), ss2sa(&state->local_saddr),
-                        &local_saddrlen) == 0)
-            local_saddrp = ss2sa(&state->local_saddr);
-
-        dispatch(state->conn->handle, local_saddrp, &conn->faddr,
+                        &local_saddrlen) < 0) {
+            krb5_klog_syslog(LOG_ERR, _("getsockname failed: %s"),
+                             error_message(errno));
+            goto kill_tcp_connection;
+        }
+        conn->local_addr.address = &conn->local_addr_buf;
+        init_addr(&conn->local_addr, ss2sa(&state->local_saddr));
```

Comment:
Hm.  I know I said to put local_addr in the connection object and not the state object, but I recently realized that init_addr() stores an alias from the socket address into the resulting krb5_fulladdr value, and it seems dangerous to have an alias from an object in conn to an object in state.  So we should either move local_saddr into conn (possibly initializing it in accept_tcp_connection()) or move local_addr into state.